### ISSUE TITLE
Template arguments in placeholder's type-constraint  are not selected correctly. 

### ISSUE THREAD

**Issue by hokein** [clangd/clangd#2033]
Given the following code:

```
template <typename T, typename K>
concept D = true;
template<typename T>
void g(D<T> auto abc) {} 
```

Go-to-definition on `D<^T> auto` will jump to the `abc`.  Hover on `T` doesn't work neither. Probably something wrong in the SelectionTree, or in the source range of the `ConceptSpecializationExpr`.

**Comment by zyn0217** [comment]
The issue is that we forget to propagate angle loci in `InventTemplateParameter`, resulting in the source range for `ConceptReference` for `D<T>` always being that of `D`, and hence our `SelectionTester::mayHit` would always return false for the selections within the argument list.

https://github.com/llvm/llvm-project/blob/3b8b1022684175e988f043f14596f2dc9b31c6c7/clang/lib/Sema/SemaType.cpp#L3101-L3133

The fix is presumably rather trivial according to other usage of `TemplateArgumentListInfo` - we just need to switch the constructor to the source location version.

**Comment by hokein** [comment]
Should be fixed by @zyn0217's commit 8070b2defa6df1f1a3f3d4ed4989047b0e1bb639
